crates/factor-outbound-redis/src/factor_redis.rs

```diff
@@ -0,0 +1,304 @@
+use anyhow::Result;
+use redis::{aio::Connection, AsyncCommands, FromRedisValue, Value};
+use spin_core::{async_trait, wasmtime::component::Resource};
+use spin_world::v1::{redis as v1, redis_types};
+use spin_world::v2::redis::{
+    self as v2, Connection as RedisConnection, Error, RedisParameter, RedisResult,
+};
+
+use tracing::{instrument, Level};
+
+struct RedisResults(Vec<RedisResult>);
+
+impl FromRedisValue for RedisResults {
+    fn from_redis_value(value: &Value) -> redis::RedisResult<Self> {
+        fn append(values: &mut Vec<RedisResult>, value: &Value) {
+            match value {
+                Value::Nil | Value::Okay => (),
+                Value::Int(v) => values.push(RedisResult::Int64(*v)),
+                Value::Data(bytes) => values.push(RedisResult::Binary(bytes.to_owned())),
+                Value::Bulk(bulk) => bulk.iter().for_each(|value| append(values, value)),
+                Value::Status(message) => values.push(RedisResult::Status(message.to_owned())),
+            }
+        }
+
+        let mut values = Vec::new();
+        append(&mut values, value);
+        Ok(RedisResults(values))
+    }
+}
+
+impl crate::InstanceState {
+    async fn is_address_allowed(&self, address: &str) -> bool {
+        // TODO: handle better
```

Comment:
Can we write a better todo here?

Reply:
This is the same `unwrap` issue I was encountering. @me-diru the way we should solve this is to just return `Result<bool>` from this function. We're okay to bubble up any parsing errors from here.